backend/src/process.py

```diff
@@ -177,9 +179,31 @@ async def __process(self, node: UsableData) -> Any:
             # Run the node and pass in inputs as args
             run_func = functools.partial(node_instance.run, *enforced_inputs)
             output = await self.loop.run_in_executor(None, run_func)
+            node_outputs = node_instance.get_outputs()
+            broadcast_data = dict()
+            if len(node_outputs) > 0:
+                if len(node_outputs) == 1:
+                    try:
+                        broadcast_data[0] = node_outputs[0].broadcast(output)
```

Comment:
If you converted `output` into an iterable if `len(node_outputs) == 1`, then you could get rid of this `if` block.

You should probably store the iterable outputs in a new variable. So something like: `output_iter = [output] if len(node_outputs) == 1 else output`.